CHANGELOG.md

```diff
@@ -0,0 +1,19 @@
+### 0.5.1 (unreleased)
+
+- New features
+  - More consistent handling of missing keys: if a config key is missing pureconfig always throws a
+    `KeyNotFoundException` now, unless the `ConfigConvert` extends the new `AllowMissingKey` trait.
+
+### 0.5.0 (Jan 3, 2017)
+
+- New features
+  - Sealed families are now converted to and from configs unambiguously by using an extra `type` field (customizable) in
+    their config representation; 
+  - New `CoproductHint` trait which allows customization of the derived `ConfigConvert` for sealed families;
+- Breaking changes
+  - The default config representation for sealed families has changed. Configs for such classes that were read
```

Comment:
I usually prefer the CHANGELOG to be concise but I can see the importance of describing what's different, in particular with a Breaking changes section. What do you think @ruippeixotog, should we make the CHANGELOG a bit more descriptive about what's going to break? I personally support @leifwickland idea.